### PR TITLE
Add functionality to plot visual extrinsics from encoder

### PR DESCRIPTION
**Pull request by dtch1997**
Changes:

- Refactor `enjoy.py` to use the same model inference code for both the base policy and the adapter
- Enable plotting of visual extrinsics by default. 

Plotting is currently implemented in a quick and hacky manner which assumes a specific concatenation order in `LocomotionVisualEncoder`. Since it does not modify the `LocomotionVisualEncoder` class it is fully backwards-compatible with previously trained agents.

### CODE REVIEW

File: enjoy.py [hunk 1]
```diff
@@ -53,6 +74,7 @@ def main():  # noqa: C901
         default=False,
         help="Do not render the environment (useful for tests)",
     )
+    parser.add_argument("--log-encoder-output", action="store_true", default=False, help="Log the encoder output to a file")
```
Comment (jennyzzt) [inline enjoy.py:77]: this parser argument is never used

**Comment by jennyzzt** [comment]
Looks good mostly. I think it would be good to plot the heightmap input too, because we want to check how the extrinsics change with heightmap.

File: enjoy.py (hunk 1 above)
Comment (dtch1997) [reply enjoy.py]: Fixed

**Comment by jennyzzt** [comment]
> > I think it would be good to plot the heightmap input too, because we want to check how the extrinsics change with heightmap.
> 
> @jennyzzt this is a bit more involved. I'm not sure how I could plot the evolution of the heightmap sensor input over time in a nice way. One idea that I have is to implement an animated plot (i.e. GIF) where each frame will contain two subplots
> 
> * on the left, an instantaneous snapshot of the heightmap input
> * on the right, the full extrinsics over time with a vertical line showing where we are in time
> 
> I'm open to suggestions as well

Maybe we can just save the heightmap data in a csv/text file. I think the current plot of the extrinsics is enough, we can just keep the heightmap data first and leave the plotting of the heightmap to later.

**Comment by dtch1997** [comment]
Sounds good. 

I will implement that, and also look into why certain tests in `test_enjoy.py` seem to be failing.

**Pull request merged**